### PR TITLE
arrow-parens: fix async case

### LINKED ISSUE #1458: Invalid violation with the arrow-parens rule when using async on a function
### Bug Report

- __TSLint version__: `3.14.0`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: `VS Code`

#### TypeScript code being linted

```ts
export const configure = async (aurelia: Aurelia) => {
    // ...
};
```

with `tslint.json` configuration:

```json
"arrow-parens": true
```

#### Actual behavior

`Parentheses are required around the parameters of an arrow function definition`

#### Expected behavior

No error/warning.

Related to #1442.

### PR DESCRIPTION
**Pull request by YuichiNukiyama**
fix #1458 

I resolve following case.

```ts
const validAsync = async (param: any) => {};
const validAsync = async (param) => {};
```

I think should warn following case, but I have no idea to validate this. 
Do you have any Idea?

`const invalidAsync = async param => {};`

### CODE REVIEW

File: test/rules/arrow-parens/test.ts.lint [hunk 1]
```diff
@@ -19,9 +19,14 @@ var barbarbar = <T>(method: (a) => T) => {
...
+const validAsync = async (param) => {};
 
 // invalid case
 var e = (a => {})(1);
          ~            [Parentheses are required around the parameters of an arrow function definition]
 var f = a => {};
         ~             [Parentheses are required around the parameters of an arrow function definition]
+
+// I think should warn following case, but I have no idea to validate this. 
+const invalidAsync = async param => {};
```
Comment (JKillian) [inline test/rules/arrow-parens/test.ts.lint]: Actually, I don't think `async param => {}` is even valid TypeScript. Can you just delete this test?

Comment (YuichiNukiyama) [inline test/rules/arrow-parens/test.ts.lint]: Really? I could compile this code. Anyway, I delete this test.

**Comment by YuichiNukiyama** [comment]
@JKillian I follow your advise.

**Comment by JKillian** [comment]
Thanks @YuichiNukiyama!

**Pull request merged**